src/Simplify.elm

```diff
@@ -5156,62 +5163,59 @@ listIntersperseCompositionChecks checkInfo =
     compositionAfterWrapIsUnnecessaryCheck listCollection checkInfo
 
 
-listAppendChecks : CheckInfo -> Maybe (Error {})
-listAppendChecks checkInfo =
-    let
-        listArgToTheLeft : Node Expression
-        listArgToTheLeft =
-            checkInfo.firstArg
-    in
+collectionAppendChecks : CollectionProperties (FromListProperties otherProperties) -> CheckInfo -> Maybe (Error {})
+collectionAppendChecks collection checkInfo =
     firstThatConstructsJust
         [ \() ->
-            case AstHelpers.getListLiteral listArgToTheLeft of
-                Just [] ->
-                    Just
-                        (alwaysReturnsLastArgError
-                            (qualifiedToString checkInfo.fn ++ " with [] as the first argument")
-                            { lastArg = secondArg checkInfo
-                            , lastArgRepresents = "second list argument"
-                            }
-                            checkInfo
-                        )
+            if collection.empty.is checkInfo.lookupTable checkInfo.firstArg then
+                Just
+                    (alwaysReturnsLastArgError
+                        (qualifiedToString checkInfo.fn ++ " " ++ descriptionForIndefinite collection.empty.description)
+                        { lastArg = secondArg checkInfo
+                        , lastArgRepresents = collection.represents
+                        }
+                        checkInfo
+                    )
 
-                _ ->
-                    Nothing
+            else
+                Nothing
         , \() ->
             case secondArg checkInfo of
-                Just listArgToTheRight ->
-                    case AstHelpers.getListLiteral listArgToTheRight of
-                        Just [] ->
-                            Just
-                                (returnsArgError (qualifiedToString checkInfo.fn ++ " with [] as the second argument")
-                                    { arg = listArgToTheLeft
-                                    , argRepresents = "first list argument"
-                                    }
-                                    checkInfo
-                                )
+                Just secondArg_ ->
+                    if collection.empty.is checkInfo.lookupTable secondArg_ then
+                        Just
+                            (Rule.errorWithFix
+                                { message = "Unnecessary append with " ++ descriptionForIndefinite collection.empty.description
```

Comment:
I'd prefer `qualifiedToString checkInfo.fn` instead of just "append". Should be more generic (e.g. if append is called differently e.g. concat or union).

Reply:
Done :+1:

Reply:
I would not just use `Tuple.second` as only the unqualified name here (which btw should probably be `AstHelpers.qualifiedName`). Why not use the qualified name?